crates/blockifier/src/execution/contract_class.rs

```diff
@@ -649,12 +649,9 @@ impl NativeContractClassV1Inner {
         // function name is what is used by Cairo Native to lookup the function.
         // Therefore it's not enough to know the function index and we need enrich the contract
         // entry point with FunctionIds from SierraProgram.
-        let lookup_fid: HashMap<usize, &FunctionId> =
-            HashMap::from_iter(sierra_program.funcs.iter().map(|fid| {
-                // This exception should never occur as the id is also in [SierraContractClass]
-                let id: usize = fid.id.id.try_into().expect("function id exceeds usize");
-                (id, &fid.id)
-            }));
+        let lookup_fid: HashMap<usize, &FunctionId> = HashMap::from_iter(
+            sierra_program.funcs.iter().enumerate().map(|(idx, func)| (idx, &func.id)),
+        );
```

Comment:
This shouldn't be a `HashMap` anymore now that the index into the vector can be used. Removing the HashMap should lead to some improvements further down as well.